src/Database/LSMTree/Internal/RawPage.hs

```diff
@@ -97,13 +103,25 @@ rawPageEntry !page !key = bisect 0 (fromIntegral dirNumKeys)
         | key == rawPageKeyAt page i   = found i
         | otherwise                    = linear (i + 1) j
 
-rawPageValue1Prefix :: RawPage -> Entry (P.Vector Word8, Word32) BlobSpan
-rawPageValue1Prefix page = case rawPageOpAt page 0 of
-  0 -> if rawPageHasBlobSpanAt page 0 == 0
-       then Insert (rawPageSingleValue page)
-       else InsertWithBlob (rawPageSingleValue page) (rawPageBlobSpanIndex page (rawPageCalculateBlobIndex page 0))
-  1 -> Mupdate (rawPageSingleValue page)
-  _ -> Delete
+-- | Lookup for a page containing one multi-page value.
+--
+-- This function does /not/ check whether the raw page contains only a single
+-- value that overflows onto multiple pages, so use with care.
+--
+-- A search through a fence-pointer index can typically tell whether a value
+-- contains a multi-page value or not, so use that before calling this function.
+rawPageValue1Prefix ::
+     RawPage
+  -> P.Vector Word8 -- ^ key
+  -> Maybe (Entry (P.Vector Word8, Word32) BlobSpan)
+rawPageValue1Prefix page !key = case compare key (rawPageKeyAt page 0) of
+    EQ -> Just $! case rawPageOpAt page 0 of
+      0 -> if rawPageHasBlobSpanAt page 0 == 0
+          then Insert (rawPageSingleValue page)
+          else InsertWithBlob (rawPageSingleValue page) (rawPageBlobSpanIndex page (rawPageCalculateBlobIndex page 0))
+      1 -> Mupdate (rawPageSingleValue page)
+      _ -> Delete
+    _ -> Nothing
```

Comment:
Small change: we still have to check whether the page contains the key we are looking for, since the compact index only tell us that key *may* be found on that page, even for multi-page values. As such, we could get false positives for intra-page lookups

A regular index (not compact) might be able to guarantee that key *is definitely* on a page when  it is a multi-page value, but this is not true for the compact index

Reply:
That's ok isn't it? We just do the lookup in the page and we find the key or not.

Reply:
Sounds good to me. Something like this?
```haskell
rawPageEntry
    :: RawPage
    -> P.Vector Word8 -- ^ key
    -> Maybe (Entry (Either (P.Vector Word8, Word32) (P.Vector Word8)) BlobSpan)
```

Reply:
Now done in #105.
```
rawPageLookup
    :: RawPage
    -> P.Vector Word8 -- ^ key
    -> RawPageLookup (Entry (P.Vector Word8) BlobSpan)

data RawPageLookup entry =
       -- | The key is not present on the page.
       LookupEntryNotPresent

       -- | The key is present and corresponds to a normal entry that fits
       -- fully within the page.
     | LookupEntryNormal !entry
     
       -- | The key is present and corresponds to an entry where the value
       -- may have overflowed onto subsequent pages. In this case the entry
       -- contains the /prefix/ of the value (that did fit on the page). The
       -- length of the suffix is returned separately.
     | LookupEntryOverflow !entry !Word32
  deriving (Eq, Functor, Show)
```